src/main/java/com/adobe/target/edge/client/model/TargetAttributesResponse.java

```diff
@@ -84,80 +115,107 @@ public TargetDeliveryResponse getResponse() {
 
   @Override
   public Map<String, Object> toMboxMap(String mbox) {
-    Map<String, Object> mboxContent = this.content.get(mbox);
-    if (mboxContent != null) {
-      return mboxContent;
-    }
-    synchronized (this.content) {
-      mboxContent = this.content.get(mbox);
-      if (mboxContent != null) {
-        return mboxContent;
-      }
-      return toMap().get(mbox);
-    }
+    return this.content.get(mbox);
   }
 
   @Override
   public Map<String, Map<String, Object>> toMap() {
-    if (this.response == null) {
+    return this.content;
+  }
+
+  private static Map<String, Map<String, Object>> toMap(TargetDeliveryResponse targetResponse) {
+    if (targetResponse == null) {
       return null;
     }
-    if (this.contentCreated) {
-      return this.content;
+
+    Map<String, Map<String, Object>> result = new HashMap<>();
+    String globalMbox = getGlobalMbox(targetResponse);
+    DeliveryResponse response = targetResponse.getResponse();
+    PrefetchResponse prefetchResponse = response.getPrefetch();
+    ExecuteResponse executeResponse = response.getExecute();
+
+    processPrefetch(result, prefetchResponse, globalMbox);
+    processExecute(result, executeResponse, globalMbox);
+
+    return toReadOnlyMap(result);
+  }
+
+  private static String getGlobalMbox(TargetDeliveryResponse response) {
+    String globalMbox = response.getResponseStatus().getGlobalMbox();
```

Comment:
`response.getResponseStatus()` can never be null right?

Reply:
Nope it is always instantiated.